### PR TITLE
Add iterator multiplexer

### PR DESCRIPTION
**Pull request by emkornfield**
This will be used in conjunction with the parallel reader in my other PR.

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      return true;
+    }
+
+    private boolean mightHaveNext() {
+      return !done.get() || !queue.isEmpty();
+    }
+
+    @Override
+    public T next() {
+      Preconditions.checkState(t != null);
```
Comment (davidrabinowitz) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java:133]: Please add error message "next element cannot be null"

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
+package com.google.cloud.bigquery.connector.common;
+
+import java.util.Iterator;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.arrow.util.Preconditions;
```
Comment (davidrabinowitz) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java:7]: Please change to the guava `Preconditions`, it has the same implementation and will make it easier to re-use the class elsewhere if needed.

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java (hunk 1 above)
Comment (emkornfield) [reply connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java]: done.

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java (hunk 2 above)
Comment (emkornfield) [reply connector/src/main/java/com/google/cloud/bigquery/connector/common/IteratorMultiplexer.java]: done.

**Comment by emkornfield** [comment]
ping @davidrabinowitz

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**